pkg/capacityscheduling/capacity_scheduling.go

```diff
@@ -605,6 +641,66 @@ func selectVictimsOnNode(
 	return victims, numViolatingVictim, true
 }
 
+// PrepareCandidate does some preparation work before nominating the selected candidate:
+// - Evict the victim pods
+// - Release the elastic quota if the victim pods reserved
+// - Reject the victim pods if they are in waitingPod map
+// - Clear the low-priority pods' nominatedNodeName status if needed
+func (c *CapacityScheduling) PrepareCandidate(candidate Candidate, fh framework.Handle, cs kubernetes.Interface, pod *v1.Pod) error {
+	for _, victim := range candidate.Victims().Pods {
+		if err := util.DeletePod(cs, victim); err != nil {
+			klog.Errorf("Error preempting pod %v/%v: %v", victim.Namespace, victim.Name, err)
+			return err
+		}
+
+		c.deletePodIfPresent(pod)
```

Comment:
Apart from this line, it is no different from upstream in `PrepareCandidate` and `getLowerPriorityNominatedPods`